src/mbgl/util/io.cpp

```diff
@@ -1,43 +1,29 @@
 #include <mbgl/util/io.hpp>
 
-#include <cstdio>
 #include <cerrno>
 #include <iostream>
-#include <sstream>
 #include <fstream>
 
 namespace mbgl {
 namespace util {
 
-void write_file(const std::string &filename, const std::string &data) {
-    FILE *fd = fopen(filename.c_str(), "wb");
-    if (fd) {
-        fwrite(data.data(), sizeof(std::string::value_type), data.size(), fd);
-        fclose(fd);
-    } else {
-        throw std::runtime_error(std::string("Failed to open file ") + filename);
-    }
-}
-
-std::string read_file(const std::string &filename) {
```

Comment:
The `read_file` → `readFile` changes replace an exception-throwing function by a function that superficially looks like it either can never fail, or will throw an exception if it fails (because it returns a `Blob`, not an `optional<Blob>`, `Result<Blob>`, or other error-indicating result type).

This a trap for new uses of `readFile`, and it also looks like many existing uses of `read_file` that have been switched to `readFile` will now silently attempt to use a falsy `Blob`, with unpredictable results, rather than unwind the stack with an exception as they used to do.

I think it's probably a design mistake for `Blob` to have an "error"/"empty" state. Especially for `readFile`, the error case needs to be called out and dealt with explicitly.